Index: TMS/src/model/CompilazioneTicket.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package model;\n\npublic class CompilazioneTicket {\n}\n
===================================================================
diff --git a/TMS/src/model/CompilazioneTicket.java b/TMS/src/model/CompilazioneTicket.java
--- a/TMS/src/model/CompilazioneTicket.java	(revision 4e7335c169d6425e0f256e9a03936367037f03de)
+++ b/TMS/src/model/CompilazioneTicket.java	(date 1768996148984)
@@ -1,4 +1,38 @@
 package model;
 
+import model.dao.TicketDao;
+import model.pojo.Ticket;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
 public class CompilazioneTicket {
+
+    private final TicketDao ticketDao;
+
+    public CompilazioneTicket(TicketDao ticketDao) {
+
+        this.ticketDao = ticketDao;
+    }
+
+
+    public void creaTicket(int idPaziente, String colore, int priorita, String sintomi) {
+
+        Ticket nuovoTicket = new Ticket();
+
+        nuovoTicket.setIdPaziente(idPaziente);
+        nuovoTicket.setColore(colore);
+        nuovoTicket.setPriorita(priorita);
+        nuovoTicket.setSintomi(sintomi);
+
+        nuovoTicket.setStato("IN_ATTESA");
+
+        LocalDateTime now = LocalDateTime.now();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+        nuovoTicket.setTimestamp(now.format(formatter));
+
+
+        ticketDao.save(nuovoTicket);
+
+
+    }
 }
Index: TMS/src/model.test/CompilazioneTicketTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>import static org.junit.jupiter.api.Assertions.*;\nclass CompilazioneTicketTest {\n  \n}
===================================================================
diff --git a/TMS/src/model.test/CompilazioneTicketTest.java b/TMS/src/model.test/CompilazioneTicketTest.java
--- a/TMS/src/model.test/CompilazioneTicketTest.java	(revision 4e7335c169d6425e0f256e9a03936367037f03de)
+++ b/TMS/src/model.test/CompilazioneTicketTest.java	(date 1768994195570)
@@ -1,4 +1,38 @@
+package model.test;
+
+import model.CompilazioneTicket;
+import model.dao.TicketDao;
+import model.pojo.Ticket;
+import org.junit.jupiter.api.Test;
+
+
 import static org.junit.jupiter.api.Assertions.*;
+
 class CompilazioneTicketTest {
-  
+
+    // Stub manuale
+    static class TicketDaoStub extends TicketDao {
+        public Ticket ticketSalvato;
+
+        @Override
+        public void save(Ticket ticket) {
+            this.ticketSalvato = ticket;
+        }
+    }
+
+    @Test
+    void testCreaTicket() {
+        // Setup
+        CompilazioneTicketTest.TicketDaoStub daoStub = new TicketDaoStub();
+        CompilazioneTicket compilazione = new CompilazioneTicket(daoStub);
+
+        // Esecuzione
+        compilazione.creaTicket(101, "Rosso", 1, "Sintomi gravi");
+
+        // Verifica
+        Ticket result = daoStub.ticketSalvato;
+
+        assertNotNull(result);
+        assertEquals(101, result.getIdPaziente());
+    }
 }
\ No newline at end of file
